### PR TITLE
Move shield definition helper functions to shield library

### LINKED ISSUE #902: Migrate JSON shape functions to shield library
There are a collection of helper functions for drawing various shapes for shields that are currently in `shield_defs.js`.  These helper functions are more appropriately moved to the shield library.

This code starts at:

https://github.com/ZeLonewolf/openstreetmap-americana/blob/main/src/js/shield_defs.js#L5-L34

### PR DESCRIPTION
**Pull request by ZeLonewolf**
Fixes #902

This PR:
1. Moves the shield definition helper functions into the shield library.
2. Completes the TypeScript definitions for the shield specification with associated refactors.
3. Adds TypeDoc generated documentation so that the new functions are documented.

Link to typedoc-generated API reference:
https://preview.ourmap.us/pr/916/shield-docs/

### CODE REVIEW

**Comment by ZeLonewolf** [comment]
Additional remaining work is to add missing typedoc documentation as noted by the typedoc warnings shown in the build output.

File: shieldlib/src/types.ts [hunk 1]
```diff
@@ -1,73 +1,209 @@
...
+  radius2?: number;
+  /** Height of diagonal edges */
+  offset?: number;
+  /** Width of the shape's outline */
+  outlineWidth?: number;
+  /** Specify whether the pointy end of the shape is on top */
+  pointUp?: boolean;
+  /** Specify whether the short side of the shape is on top */
+  shortSideUp?: boolean;
+  /** Specify the angle of the defining vertex of the shape */
```
Comment (claysmalley) [inline shieldlib/src/types.ts:97]:
```suggestion
  /** Specify the angle at which the sides of the shape deviate from vertical. Higher number means pointier sides */
```

File: shieldlib/src/types.ts [hunk 2]
```diff
@@ -1,73 +1,209 @@
...
+  fillColor: string;
+  /** Stroke (border) color */
+  strokeColor: string;
+  /** Width of the shape */
+  rectWidth?: number;
+  /** Radius of the shape's corners */
+  radius?: number;
+  /** Radius of the shapes's first corner. This is used for shapes that can specify multiple raidus values */
+  radius1?: number;
+  /** Radius of the shapes's second corner. This is used for shapes that can specify multiple raidus values */
```
Comment (claysmalley) [inline shieldlib/src/types.ts:87]:
```suggestion
  /** Radius of the shape's first corner. This is used for shapes that can specify multiple radius values */
  radius1?: number;
  /** Radius of the shape's second corner. This is used for shapes that can specify multiple radius values */
```

File: shieldlib/src/types.ts [hunk 3]
```diff
@@ -1,73 +1,209 @@
...
+  radius2?: number;
+  /** Height of diagonal edges */
```
Comment (claysmalley) [inline shieldlib/src/types.ts:89]:
```suggestion
  /** Distance from top or bottom edge to vertices. Higher number means pointier top and/or bottom */
```

**Review by claysmalley — COMMENTED** [review]
I guess this is a good time to consider renaming some of the shapeBlank parameters. I've never liked `angle` and `offset`. Perhaps `sideAngle` and `yOffset` are a little more descriptive?

**Comment by ZeLonewolf** [comment]
> I guess this is a good time to consider renaming some of the shapeBlank parameters. I've never liked `angle` and `offset`. Perhaps `sideAngle` and `yOffset` are a little more descriptive?

Yes, that sounds fine.  However, I'm going to convert the remaining draw code to typescript to ensure that I don't miss any when I do the rename.

**Comment by ZeLonewolf** [comment]
`yOffset` and `sideAngle` are now implemented, e.g.:
https://preview.ourmap.us/pr/916/shield-docs/functions/pentagonUpShield.html

**Review by claysmalley — COMMENTED** [review]
Let's go ahead and propagate `angle` → `sideAngle` and `offset` → `yOffset`, as well as their associated documentation changes, to the rest of the codebase. 

Other than that, looks good. I don't know much about TypeScript, so a once-over by someone with more familiarity would be helpful.

File: shieldlib/src/shield_helper.ts [hunk 4]
```diff
@@ -0,0 +1,757 @@
...
+export function octagonVerticalShield(
+  yOffset: number,
+  angle: number,
+  fillColor: string,
+  strokeColor: string,
+  textColor: string,
+  radius: number,
+  rectWidth: number
+): ShieldDefinition {
+  let angleInRadians = (angle * Math.PI) / 180;
```
Comment (claysmalley) [inline shieldlib/src/shield_helper.ts:633]:
```suggestion
 * @param {*} sideAngle - Angle (in degrees) at which sides deviate from vertical
 * @param {*} fillColor - Color of octagon background fill
 * @param {*} strokeColor - Color of octagon outline stroke
 * @param {*} textColor - Color of text (defaults to strokeColor)
 * @param {*} radius - Corner radius of octagon (defaults to 2)
 * @param {*} rectWidth - Width of octagon (defaults to variable-width)
 * @returns a shield definition object
 */
export function octagonVerticalShield(
  yOffset: number,
  sideAngle: number,
  fillColor: string,
  strokeColor: string,
  textColor: string,
  radius: number,
  rectWidth: number
): ShieldDefinition {
  let angleInRadians = (sideAngle * Math.PI) / 180;
```

**Comment by claysmalley** [comment]
`package-lock.json` has changed, apparently. Needs to be committed.

**Review by claysmalley — APPROVED** [review]
Looks good to me. Could still use a proofread by someone more versed with TypeScript.

**Comment by ZeLonewolf** [comment]
Any outstanding typescript issues can be addressed in future PRs.

**Pull request merged**